src/components/Route/EditRoute.tsx

```diff
@@ -0,0 +1,219 @@
+                    <input className='hidden-input' type='file' id="file" accept='image/*'
+                      onChange={handleFileSelect} />
+                    <label htmlFor="file" className='native-button'>Replace photo</label>
+                  </HStack> 
+                </>
+                
+              }
+
+            </>
+            :
+            // show thumbnail preview and remove button
+            <>
+              {thumbnailFile !== undefined ?
+                <>
+                  {imageCompressing ? 
+                    <Text variant='subtext'>Compressing image...</Text>
+                    :
+                    <>
+                      <img src={URL.createObjectURL(thumbnailFile)} alt='thumbnail' className='route-popup-avatar'/>
+                      <button className='native-button' onClick={() => {
+                        setShowStoredThumbnail(true);
+                        setThumbnailFile(undefined);
+                      }}>
+                        Reset to stored thumbnail
+                      </button>
+                    </>
+                  }
+                </>
+                :
+                <>
+                  <input className='hidden-input' type='file' id="file" accept='image/*'
+                    onChange={handleFileSelect} />
+                  <label htmlFor="file" className='native-button'>Select photo</label>
+                </>
+              }
+            </>
+        }
+        <HStack alignItems='center'space={1}width='100%'>
+          <Text bold >Description: </Text>
+          <Input type='text' value={description} onChangeText={setDescription} width='75%' 
+            multiline numberOfLines={2} />
+        </HStack>
+        <HStack alignItems='center'space={1}width='100%'>
+          <Text bold >Setter: </Text>
+          <Radio.Group name='Setter' defaultValue='User' onChange={(nextVal) => setSetterType(nextVal)}>
+            <HStack alignItems='center'space={2}>
+              <Radio value='User'>User</Radio>
+              <Radio value='Custom'>Custom</Radio>
+            </HStack>
+          </Radio.Group>
+        </HStack>
+        {
+          setterType === 'User' ?
+            <Text>This will be a user search component</Text>
+            :
+            <Input 
+              type='text' 
+              value={setterRawName} 
+              placeholder='Non-user setter' 
+              onChangeText={setSetterRawName} 
+              width='50%' 
+            />
+        }
+        <HStack>
+          <Text bold >Natural Rules: </Text>
+          <Select placeholder='Natural Rules' onValueChange={(rulesString) => {
+            const rules = rulesString as NaturalRules;
+            setNaturalRules(rules);
+          }} defaultValue={route.naturalRules}>
+            {Object.values(NaturalRules).map((rules) => {
+              return <Select.Item key={rules} label={rules} value={rules} />;
+            })}
+          </Select>
+        </HStack>
```

Comment:
looks great, only small thing I noticed was that the Natural Rules text isn't aligned the same as the others
`<HStack alignItems='center'space={1}width='100%'>` on line 168